src/Core/Models/Data/EventItem.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Text.Json.Serialization;
+using Bit.Core.Enums;
+
+namespace Bit.Core.Models.Data;
+
+public class EventItem : IEvent
+{
+    public EventItem() {}
+
+    public EventItem(IEvent e)
+    {
+        Id = Guid.NewGuid().ToString();
+        Date = e.Date;
+        Type = e.Type;
+        UserId = e.UserId;
+        OrganizationId = e.OrganizationId;
+        InstallationId = e.InstallationId;
+        ProviderId = e.ProviderId;
+        CipherId = e.CipherId;
+        CollectionId = e.CollectionId;
+        PolicyId = e.PolicyId;
+        GroupId = e.GroupId;
+        OrganizationUserId = e.OrganizationUserId;
+        ProviderUserId = e.ProviderUserId;
+        ProviderOrganizationId = e.ProviderOrganizationId;
+        DeviceType = e.DeviceType;
+        IpAddress = e.IpAddress;
+        ActingUserId = e.ActingUserId;
+        SystemUser = e.SystemUser;
+        DomainName = e.DomainName;
+        SecretId = e.SecretId;
+        ServiceAccountId = e.ServiceAccountId;
+    }
+
+    [JsonPropertyName("id")]
+    public string Id { get; set; }
+
+    [JsonPropertyName("type")]
+    public EventType Type { get; set; }
+    [JsonPropertyName("ip")]
```

Comment:
🎨 Trying to save space with some of these? I would think we'll never need to look at the DB and always use this mapping so maybe even more compressed? Reminds me of a JWT token.